test/functional/options/num_options_spec.lua

```diff
@@ -0,0 +1,42 @@
+-- Tests for :setlocal and :setglobal
+
+local helpers = require('test.functional.helpers')(after_each)
+local clear, execute, eval, eq, nvim =
+  helpers.clear, helpers.execute, helpers.eval, helpers.eq, helpers.nvim
+
+local function get_num_option_global(opt)
+  return nvim('command_output', 'setglobal ' .. opt .. '?'):match('%d+')
```

Comment:
This is unfortunate. I noticed that `nvim_get_option` API function doesn't return the global value of a local option. Need to fix that...